packages/flutter_test/lib/src/finders.dart

```diff
@@ -469,3 +483,24 @@ class _ElementPredicateFinder extends MatchFinder {
     return predicate(candidate);
   }
 }
+
+class _DescendantFinder extends Finder {
+  _DescendantFinder(this.ancestor, this.descendant, { bool skipOffstage: true }) : super(skipOffstage: skipOffstage);
+
+  final Finder ancestor;
+  final Finder descendant;
+
+  @override
+  String get description => '${ancestor.description} has descendant(s) with ${descendant.description} ';
+
+  @override
+  Iterable<Element> apply(Iterable<Element> candidates) {
+    return candidates.where((Element element) => descendant.evaluate().contains(element));
+  }
+
+  @override
+  Iterable<Element> get _allElements {
```

Comment:
overriding a private is sketchy. it's not part of the API you're expected to extend.